### PR TITLE
Implemented :meth:`.CoordinateSystem.input_to_graph_coords` and fixed :meth:`.CoordinateSystem.angle_of_tangent`

### LINKED ISSUE #2023: Typo in coordinate_systems
At `manim/mobject/coordinate_systems.py` line 195 and 196, the origin code is 

```python
    def get_y_unit_size(self):
        return self.get_x_axis().get_unit_size()
```

And it should be
```python
    def get_y_unit_size(self):
        return self.get_y_axis().get_unit_size()
```
This typo is small but may cause confusion if someone use this method. Please fix it soon.

### PR DESCRIPTION
**Pull request by icedcoffeeee**
## Overview: What does this pull request change?

Fixes #2023 and #1701 
## Motivation and Explanation: Why and how do your changes improve the library?

Before:
```py
>>> from manim import *
Manim Community v0.10.0

>>> ax = Axes([-2.5, 2.5], [-1, 5], x_length=10, y_length=6)
>>> graph = ax.get_graph(lambda x: x ** 2, x_range=[-2.5,2.5,0.01])
>>> ax.angle_of_tangent(2, graph)
1.1071487177940904
>>> np.arctan(4)
1.3258176636680326
```
After:
```py
>>> from manim import *
Manim Community v0.9.0

>>> ax = Axes([-2.5, 2.5], [-1, 5], x_length=10, y_length=6)
>>> graph = ax.get_graph(lambda x: x ** 2, x_range=[-2.5,2.5,0.01])
>>> ax.angle_of_tangent(2, graph)
1.3258176636680326
>>> np.arctan(4)
1.3258176636680326
```
## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by Darylgolden — APPROVED** [review]
LGTM, but perhaps it would be slightly better designed if there were a `input_to_graph_coords` method, which the angle of tangent can be calculated directly from?

**Comment by icedcoffeeee** [comment]
@Darylgolden That's a great idea. I updated the methods. :)

File: manim/mobject/coordinate_systems.py [hunk 1]
```diff
@@ -765,6 +765,19 @@ def construct(self):
                     f"x={x} not located in the range of the graph ([{self.p2c(graph.get_start())[0]}, {self.p2c(graph.get_end())[0]}])",
                 )
 
+    def input_to_graph_coords(self, x: float, graph: "ParametricFunction") -> Tuple:
+        """
+        Returns a tuple of the axis relative coordinates of the point
+        on the graph based on the x-value given.
+        """
+        return x, graph.underlying_function(x)
```
Comment (hydrobeam) [inline manim/mobject/coordinate_systems.py:773]: Could you add an example (doesn't need to be visual)  + more docs for the parameters of this method?, it's a little confusing how this is functionally different from `i2gp`. Also, I'm a little unsure of adding a public method which doesn't really serve a unique purpose 🤔

Comment (icedcoffeeee) [reply manim/mobject/coordinate_systems.py:773]: > I'm a little unsure of adding a public method which doesn't really serve a unique purpose

I sort of see what you mean, but I also do see myself using it, like passing this method into `Tex` or `Text` to label a coordinate.

File: manim/mobject/coordinate_systems.py [hunk 2]
```diff
@@ -814,6 +814,29 @@ def construct(self):
...
+
+        Examples
+        --------
+
+        .. code-block:: python
+
+            ax = Axes()
+            curve = ax.get_graph(lambda x: x ** 2)
+            ax.input_to_graph_coords(x=3, graph=curve)
+            # (3, 9)
```
Comment (behackl) [inline manim/mobject/coordinate_systems.py:830]: Let's turn this into a proper doctest to improve test coverage. Also `get_graph` -> `plot`.

```suggestion
            >>> from manim import Axes
            >>> ax = Axes()
            >>> parabola = ax.plot(lambda x: x ** 2)
            >>> ax.input_to_graph_coords(x=3, graph=parabola)
            (3, 9)
```

**Review by behackl — APPROVED** [review]
We should include this PR in the release as well; have slightly modified the given example so that it is tested when the documentation is built in our pipeline.

File: manim/mobject/coordinate_systems.py [hunk 3]
```diff
@@ -943,6 +943,30 @@ def construct(self):
...
 
+    def input_to_graph_coords(self, x: float, graph: "ParametricFunction") -> Tuple:
+        """
+        Returns a tuple of the axis relative coordinates of the point
+        on the graph based on the x-value given.
+
+        Examples
+        --------
+
+        .. code-block:: python
```
Comment (behackl) [inline manim/mobject/coordinate_systems.py:954]:
```suggestion
        .. code-block:: pycon
```

**Pull request merged**